phylonco-lphy/src/main/java/phylonco/lphy/evolution/copynumbermodel/CopyNumberCompressor.java

```diff
@@ -0,0 +1,289 @@
+
+            while ((line = reader.readLine()) != null) {
+                line = line.trim();
+                if (line.isEmpty()) continue;
+
+                try {
+                    String[] parts = line.split("\t");
+                    if (parts.length < 3 + numCells) {
+                        System.err.println("Warning: Line " + lineNumber + " has insufficient columns, skipping");
+                        continue;
+                    }
+
+                    String chr = parts[0].trim();
+                    long start = Long.parseLong(parts[1].trim());
+                    long end = Long.parseLong(parts[2].trim());
+
+                    int[] copyNumbers = new int[numCells];
+                    for (int i = 0; i < numCells; i++) {
+                        copyNumbers[i] = Integer.parseInt(parts[3 + i].trim());
+                    }
+
+                    rows.add(new CNVRow(chr, start, end, copyNumbers));
+
+                } catch (NumberFormatException e) {
+                    System.err.println("Warning: Invalid number format at line " + lineNumber + ", skipping");
+                }
+
+                lineNumber++;
+            }
+        }
+
+        return rows;
+    }
+
+    /**
+     * Write compressed results to file in GINKGO format
+     */
+    public static void writeCompressedData(List<CompressedSegment> segments,
+                                           String outputPath, String[] cellNames) throws IOException {
+        try (PrintWriter writer = new PrintWriter(new FileWriter(outputPath))) {
+            // Write header in GINKGO format: Chr Start End Cell.1 Cell.2 Cell.3 ...
+            writer.print("Chr\tStart\tEnd");
+            for (String cellName : cellNames) {
+                writer.print("\t" + cellName);
+            }
+            writer.println();
+
+            // Write compressed segments in GINKGO format
+            for (CompressedSegment segment : segments) {
+                writer.println(segment.toOutputString());
+            }
+        }
+    }
+
+    /**
+     * Extract cell names from header
+     */
+    public static String[] extractCellNames(String filePath) throws IOException {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String headerLine = reader.readLine();
+            if (headerLine == null) {
+                throw new IOException("File is empty");
+            }
+
+            String[] headers = headerLine.split("\t");
+            String[] cellNames = new String[headers.length - 3];
+
+            for (int i = 3; i < headers.length; i++) {
+                cellNames[i - 3] = headers[i];
+            }
+
+            return cellNames;
+        }
+    }
+
+    /**
+     * Main method
+     */
+    public static void main(String[] args) {
+        //file path
+        String inputFile = "/Users/ming/Desktop/Aug22/CNV/2nd/CRC01-SegCopy_filtered_onlyPM.txt";
```

Comment:
We should use a relative path here, and I suggest using a file name like "copyNumberSim.txt" or similar with dummy data.